cli/tests/functional/src/test/java/io/helidon/tests/functional/TestUtils.java

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.tests.functional;
+
+import io.helidon.build.common.FileUtils;
+import io.helidon.build.common.OSType;
+import io.helidon.build.common.maven.MavenCommand;
+import io.helidon.build.common.maven.MavenVersion;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ServerSocket;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class TestUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(TestUtils.class.getName());
+
+    static void downloadMavenDist(Path destination, String version) throws IOException {
+
+        Path zipPath = destination.resolve("maven-" + version + ".zip");
+        URL mavenUrl = new URL(
+                String.format("http://archive.apache.org/dist/maven/maven-3/%s/binaries/apache-maven-%s-bin.zip", version, version)
+        );
+
+        LOGGER.info("Downloading maven from URL : " + mavenUrl);
+
+        downloadFileFromUrl(zipPath, mavenUrl);
+
+        LOGGER.info("Maven download done.");
+        LOGGER.info("Unzip Maven started ...");
+
+        FileUtils.unzip(zipPath, destination);
+
+        LOGGER.info("Unzip Maven done.");
+
+        Files.walk(destination)
+                .map(Path::toFile)
+                .filter(File::isFile)
+                .forEach(file -> file.setExecutable(true));
+
+        FileUtils.delete(zipPath);
+
+        Files.walk(destination)
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .filter(name -> name.equals("mvn"))
+                .findFirst().orElseThrow();
+    }
+
+    static void downloadFileFromUrl(Path destination, URL url) {
+        try {
+            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("www-proxy.us.oracle.com", 80));
```

Comment:
Please don't hard-code proxy or any internal URL.
Take a look at `Proxy.setProxyPropertiesFromEnv`, since env variables are propagated implicitly to forked processes, you will get those without any configuration.
Also take a look at the class called `NetworkConnection` it includes retries and a few other goodies.

Reply:
Thanks for pointing me to the right classes, I made the required changes.